product-service/src/requestHandlers/getByID.js

```diff
@@ -0,0 +1,21 @@
+import products from '../data/products.json';
+import { getHeaders, generateErrorResponse } from '../helpers/index.js'
+
+export const getProductById = async event => {
+    try {
+        const id = event.pathParameters.productId;
+        const product = products.find(a => a.id === id);
+        if (!product) {
+            throw new Error("Product not found");
```

Comment:
It not a good practice to use `throw` as `goto` statement. Here you check that `product` exist and you know the expected behavior. You can simply `return generateErrorResponse(new Error("Product not found"), 404)`